### DISCUSSION TITLE
Replace deprecated "set-env"

### DISCUSSION THREAD

**Discussion by Benwick91** [community/community#26345] [Actions]
<p>Dear Community,</p>
<p>I am using deprecated “set-env” and now I have to fix it.</p>
<p>I am using it like this:</p>
<pre><code class="lang-auto">            - name: Setup
              run: |
                  echo "::set-env name=version_prefix::# ${release_version}"
                  echo "::set-env name=file_content::${changelog_content}"
              env:
                  release_version: ${{ steps.release_drafter.outputs.name }}
                  changelog_content: ${{ steps.read-changelog.outputs.content }}

            - name: Update Log
              if: startsWith(env.file_content, env.version_prefix) != true
              run: echo -en "# $release_version\n$changes\n$(cat $changelog_filepath)" &gt;$changelog_filepath
              env:
                  changelog_filepath: ${{ github.workspace }}/changelog/CHANGELOG.md
                  release_version: ${{ steps.release_drafter.outputs.name }}
                  changes: ${{ steps.release_drafter.outputs.body }}
</code></pre>
<p>What is the new alternative for that?</p>
<p>Thank you!</p>

**Comment by otter-computer** [comment, marked as answer]
<p>This has been replaced by passing variables into <code>$GITHUB_ENV</code>.</p>
<aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">
      <a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-commands-for-github-actions#environment-files" target="_blank" rel="noopener">docs.github.com</a>
  </header>
  <article class="onebox-body">
    <img src="" class="thumbnail" width="16" height="16">

<h3><a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-commands-for-github-actions#environment-files" target="_blank" rel="noopener">Workflow commands for GitHub Actions - GitHub Docs</a></h3>



  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>For example:</p>
<pre><code class="lang-auto">echo "version_prefix=${release_version}" &gt;&gt; $GITHUB_ENV
</code></pre>
<p>The only thing I believe you’ll need to change in your workflow is <code>echo</code> commands and convert them to the new syntax. Be aware that this may be slightly different if your strings are multiline.</p>

**Comment by Benwick91** [comment]
<p>Edit: Thank you! It works. I was using the delimiter the wrong way. Now its fine. Thank you!</p>
<hr>
<p>I tried this, but the action fails:</p>
<pre><code class="lang-auto"> Error: Unable to process file command 'env' successfully.
Error: Invalid environment variable format '## Bugs'
</code></pre>
<p>The strings are multiline.</p>

**Comment by brettcurtis** [comment]
<p><a class="mention" href="/benwick91">@benwick91</a> can you share? I’m having trouble with this. My old env:</p>
<pre><code class="lang-auto">        env:
          PLAN: "terraform\n${{ steps.plan.outputs.stdout }}"
</code></pre>
<p>If tried the new way like so:</p>
<pre><code class="lang-auto">      - name: Environment Variables
        run: |
          echo -e "PLAN=terraform\n${{ steps.plan.outputs.stdout }}" &gt;&gt; ${GITHUB_ENV}
</code></pre>

**Comment by Benwick91** [comment]
<p>I am using it this way:</p>
<pre><code class="lang-auto">                  echo "version_prefix=# ${release_version}" &gt;&gt; $GITHUB_ENV
                  echo "file_content&lt;&lt;{delimiter}
                  ${changelog_content}
                  {delimiter}" &gt;&gt; $GITHUB_ENV
</code></pre>
<p>Your way seems to be right and the you sould can use it with <code>${PLAN}</code>.</p>
<p>But in your old env you doesn’t use <code>set-env</code>. Can’t you use it the old way any more?</p>
<p>I hope this helps :slight_smile:</p>

**Comment by brettcurtis** [comment]
<p><a class="mention" href="/benwick91">@benwick91</a> - yeah I guess the old way is still fine. I was under the assumption that it was the same as set-env and being deprecated. :man_facepalming:</p>
